sources/cfsignal/src/cloudformation.rs

```diff
@@ -0,0 +1,55 @@
+use crate::error::{self, Result};
+use imdsclient::ImdsClient;
+use log::info;
+use rusoto_cloudformation::{CloudFormation, CloudFormationClient, SignalResourceInput};
+use rusoto_core::Region;
+use snafu::{OptionExt, ResultExt};
+use std::str::FromStr;
+
+pub async fn signal_resource(
+    stack_name: String,
+    logical_resource_id: String,
+    status: String,
+) -> Result<()> {
+    info!("Connecting to IMDS");
+    let mut client = ImdsClient::new();
+    let instance_id = get_instance_id(&mut client).await.unwrap();
+    let region = get_region(&mut client).await.unwrap();
+
+    info!(
+        "Region: {:?} - InstanceID: {:?} - Signal: {:?}",
+        region, instance_id, status
+    );
+
+    let client = CloudFormationClient::new(Region::from_str(&region).unwrap());
+    let signal_resource_input = SignalResourceInput {
+        stack_name,
+        logical_resource_id,
+        status,
+        unique_id: instance_id,
+    };
+
+    client.signal_resource(signal_resource_input).await.context(error::SignalResource);
```

Comment:
Need to handle the error. This error eventually floats up to main and gets printed out
```suggestion
    client.signal_resource(signal_resource_input).await.context(error::SignalResource)?;
```